xdem/coreg/biascorr.py

```diff
@@ -219,11 +234,26 @@ def _fit_biascorr(  # type: ignore
                     "with function {}.".format(", ".join(list(bias_vars.keys())), self._meta["fit_func"].__name__)
                 )
 
+            if isinstance(diff, np.ndarray):
+                ydata = diff[subsample_mask].flatten()
+                xdata = np.array([var[subsample_mask].flatten() for var in bias_vars.values()]).squeeze()
+                # TODO - there is a bug here still
+                # sigma = (weights[subsample_mask].flatten() if weights is not None else None,)
+                sigma = None
+            elif isinstance(diff, da.Array):
+                ydata = diff.vindex[subsample_mask].flatten().compute()  # type:ignore [assignment]
+                xdata = [var.vindex[subsample_mask].flatten() for var in bias_vars.values()]  # type:ignore [assignment]
```

Comment:
Don't you need a `.compute()` here as well?

Reply:
Yes, even if the `bias_vars` are not delayed (no operation since their creation/passing in the `_fit_rst_rst`) function, the compute seems needed, otherwise you get a dask array output that won't be understood by the following steps:
```python
da.ones((10000, 10000)).vindex[[0, 1], [0, 1]]
Out[12]: dask.array<vindex-merge, shape=(2,), dtype=float64, chunksize=(2,), chunktype=numpy.ndarray>
da.ones((10000, 10000)).vindex[[0, 1], [0, 1]].compute()
Out[13]: array([1., 1.])
```

Reply:
thank you! nice catch :) - changed it!